.github/workflows/cross-arch-test.yaml

```diff
@@ -0,0 +1,124 @@
+---
+name: Cross-Platform Build Tests
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build-aix:
+    strategy:
+      matrix:
+        goarch: [ppc64]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      - id: goversion
+        run: echo "goversion=$(cat .go-version)" >> "$GITHUB_OUTPUT"
+      - uses: actions/setup-go@d35c59abb061a4a6fb18e82ac0862c26744d6ab5 # v5.5.0
+        with:
+          go-version: ${{ steps.goversion.outputs.goversion }}
+
+      - name: Build for aix/${{ matrix.goarch }}
+        run: |
+          GOOS=aix GOARCH=${{ matrix.GOARCH }} go build ./...
+
+  build-android:
+    strategy:
+      matrix:
+        goarch: [arm64]
```

Comment:
Hi @ahrtr 

So this job has taken long time, due to compiler issues 

Android is supported by the following Archs

```
 go tool dist list | grep android/               
android/386
android/amd64
android/arm
android/arm64
```

However, running all of them is quiet impossible, due to many reasons, you can see the CI failures. 

I have tried using many if/else and check for the Arch, but in the end i hit the Android compiler wall

Most of the errors in the end required build tags to proceed.

What I could do however, to add a step to download the Android compiler, and run it on `arm64` to prove it compiles. 

I hope this is sufficient 👍🏽

Reply:
i really tried this initially, but did not work 

let me try again now 

also this linter [issue](https://github.com/etcd-io/bbolt/actions/runs/15740411604/job/44364057092?pr=986) is new, i dont know how 🤔